src/lib.rs

```diff
@@ -91,6 +91,10 @@ pub mod windows {
     //! Windows only extensions.
 
     pub use crate::sys::named_pipe::NamedPipe;
+    // blocking windows uds which mimick std implementation used for tests
+    cfg_net! {
+        pub use crate::sys::windows::stdnet;
+    }
```

Comment:
We're not going to expose this.

Reply:
It has previously been stated the tests shouldn't change. To make the tests not change we need to have a blocking windows UDS implementation. Having this requirement only leaves us 2 options.

- 1: Add ``#[cfg(test)]`` to the expose so it can only be used in tests
- 2: Implement UDS support in the rust STD

Would option 1 even be allowed? or an Option?

If it isn't an option then if we want windows UDS in tokio there is a requirement for it being implemented in the STD First. Originally it was thought that we could add Windows UDS since MIO already supports Windows Named Pipes without std support.

So is the president that std support is required. That is fine, it is nice to just know that is definitly the step forward. Because the only non-std way to fulfill the no-changing-test-requirement would be option 1.

Reply:
> It has previously been stated the tests shouldn't change.

Yes, we need to ensure that the existing behaviour works on Windows as well.

> To make the tests not change we need to have a blocking windows UDS implementation. Having this requirement only leaves us 2 options.
> 
> * 1: Add `#[cfg(test)]` to the expose so it can only be used in tests
> * 2: Implement UDS support in the rust STD
> 
> Would option 1 even be allowed? or an Option?

The first option doesn't work as we use Rust's integration tests, which doesn't have access to another crate's `#[cfg(test)]` exposed items. The second option would be ideal, but that's currently not possible as far as I know.

> If it isn't an option then if we want windows UDS in tokio there is a requirement for it being implemented in the STD First. Originally it was thought that we could add Windows UDS since MIO already supports Windows Named Pipes without std support.
> 
> So is the precedent that std support is required. That is fine, it is nice to just know that is definitly the step forward. Because the only non-std way to fulfill the no-changing-test-requirement would be option 1.

You're overlooking a third option. Have a blocking implementation inside the test code.